tests/components/laundrify/test_config_flow.py

```diff
@@ -0,0 +1,193 @@
+)
+
+from . import (
+    _patch_laundrify_exchange_code,
+    _patch_laundrify_get_account_id,
+    _patch_laundrify_get_machines,
+    _patch_laundrify_validate_token,
+    create_entry,
+)
+from .const import VALID_ACCESS_TOKEN, VALID_AUTH_CODE, VALID_USER_INPUT
+
+
+def _patch_setup_entry():
+    return patch("homeassistant.components.laundrify.async_setup_entry")
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    with _patch_laundrify_exchange_code(), _patch_laundrify_get_account_id(), _patch_setup_entry() as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["errors"] is None
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input=VALID_USER_INPUT,
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == DOMAIN
+    assert result["data"] == {
+        CONF_ACCESS_TOKEN: VALID_ACCESS_TOKEN,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_format(hass: HomeAssistant) -> None:
+    """Test we handle invalid format."""
+    with _patch_laundrify_exchange_code() as laundrify_mock:
+        laundrify_mock.side_effect = exceptions.InvalidFormat
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={CONF_SOURCE: SOURCE_USER},
+            data={CONF_CODE: "invalidFormat"},
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_CODE: "invalid_format"}
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    with _patch_laundrify_exchange_code() as laundrify_mock:
+        laundrify_mock.side_effect = exceptions.UnknownAuthCode
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={CONF_SOURCE: SOURCE_USER},
+            data=VALID_USER_INPUT,
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_CODE: "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass: HomeAssistant):
+    """Test we handle cannot connect error."""
+    with _patch_laundrify_exchange_code() as laundrify_mock:
+        laundrify_mock.side_effect = exceptions.ApiConnectionException
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={CONF_SOURCE: SOURCE_USER},
+            data=VALID_USER_INPUT,
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_unkown_exception(hass: HomeAssistant):
+    """Test we handle all other errors."""
+    with _patch_laundrify_exchange_code() as laundrify_mock:
```

Comment:
same comment as above. `yield laundrify_mock` in your fixture, and if the fixture is called `my_fixture`, you can say `my_fixture.side_effect = Exception`